gateway/src/main/java/io/camunda/zeebe/gateway/RequestMapper.java

```diff
@@ -255,4 +267,30 @@ public static DirectBuffer ensureJsonSet(final String value) {
       }
     }
   }
+
+  public static String ensureTenantIdSet(
+      final String requestName, final String tenantId, final boolean isMultiTenancyEnabled) {
+
+    final boolean hasTenantId = StringUtils.hasText(tenantId);
+    if (isMultiTenancyEnabled) {
+
+      if (!hasTenantId) {
+        throw new InvalidTenantRequestException(requestName, "Client didn't provide a tenant ID.");
+      }
+
+      if (!TENANT_ID_MASK.matcher(tenantId).matches()) {
+        throw new InvalidTenantRequestException(
+            requestName, "Client sent a tenant ID which includes invalid characters.");
```

Comment:
❌ It's not always just invalid characters. It could also be that it's too long which isn't conveyed by the message

Reply:
I'll add a separate case/error message when the length is exceeded.